src/Cardano/Crypto/Wallet.hs

```diff
@@ -75,7 +76,7 @@ data XPub = XPub !Edwards25519.PointCompressed !ChainCode
 instance NFData XPub
 instance Hashable XPub
 
-newtype XSignature = XSignature Edwards25519.Signature
+newtype XSignature = XSignature { unXSignature :: Edwards25519.Signature }
```

Comment:
here I don't want to expose the inner of Signature, because it's likely to change type really soon after moving back to the C implementation; so I'ld rather have a smart constructor that take binary and a deconstructor that generate bytestring (or something else more convenient). Also it could be just an instance of `ByteArrayAccess` that allow convertion between bytestring types.